### PR TITLE
Proposed changes to better handle Buchla ranges and unipolarity in CV-generating apps.

### PR DESCRIPTION
**Pull request by mathomas**
Here's another batch of changes you may find of use.  I fully admit to getting to these changes empirically, using the 'scope and iterative testing, rather than via a deep understanding of the various ways to set voltages/octaves in the codebase.  Basically, I iterated until there was no clipping of the lower half of the waveforms due to insufficient offset, and no overshoot on the high side of the waveform.

One thing I notice is the overall MAX_CV constant needs to be 10v (which makes sense), or the Vector stuff ends up getting clipped (or my measurements were bad).  But then the AD EG overshoots by about 1.5v (up to 11.5-ish).  There seems to be some non-orthogonality going on, but I didn't spend as much time debugging that as I possibly could/should have.  I don't think the overshoot is harmful, but it bugs me all the same.

### CODE REVIEW

File: software/o_c_REV/HEM_VectorEG.ino [hunk 1]
```diff
@@ -164,8 +164,12 @@ private:
         osc[ch] = WaveformManager::VectorOscillatorFromWaveform(waveform);
         waveform_number[ch] = waveform;
         osc[ch].SetFrequency(freq[ch]);
+#ifdef BUCHLA_4U
```
Comment (mathomas) [inline software/o_c_REV/HEM_VectorEG.ino]: I wonder if a global #define for (12 << 7) makes sense?

File: software/o_c_REV/HemisphereApplet.h [hunk 2]
```diff
@@ -27,7 +27,11 @@
 
 #define LEFT_HEMISPHERE 0
 #define RIGHT_HEMISPHERE 1
+#ifdef BUCHLA_4U
```
Comment (mathomas) [inline software/o_c_REV/HemisphereApplet.h]: Back to a 10v max value, but does lead to overshoot in AD EG app.  If I scale back here, that affects other apps.  Living with overshoot instead.

**Comment by Chysn** [comment]
Thanks again! I'm reviewing these now. I appreciate your work a lot, as I wasn't sure how to approach Buchla updates without having any Buchla equipment!

**Comment by mathomas** [comment]
> Thanks again! I'm reviewing these now. I appreciate your work a lot, as I wasn't sure how to approach Buchla updates without having any Buchla equipment!

My pleasure.  THANK YOU for Hemisphere Suite.  It’s quite an achievement, and my hat is off to you.

I’ve been an OC user for a couple of years (bought one of the very first ones).  I first tried HS in one of my Eurorack OCs back in November, but my Buchla came along pretty quickly behind those first experiences.  I bought a 2OC before my Skylab even arrived, because I knew I’d need OC and Hemisphere functionality in Buchla-land.  I’m just glad I am a developer and could at least try to contribute a little (even though my C++ days are 25 years behind me — thankfully your code is easy to follow and these changes are simple).

I’m sure there are some lagging apps that still need addressing (like Lower Rents seems to clip or wrap on my scope, in XY mode, anyway, but I haven’t figured out why yet — probably user error).  Still, these changes make HS useful to me in Buchla-land, and therefore I hope they are useful to others as well.

**Pull request merged**